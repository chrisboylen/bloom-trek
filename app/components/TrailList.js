import React from 'react';
import { ScrollView, StyleSheet, TouchableOpacity } from 'react-native';
import { List, ListItem } from 'react-native-elements';
import PropTypes from 'prop-types';

const TrailList = ({ trails, handleSelectedTrail }) => (
  <ScrollView>
    <List containerStyle={styles.list}>
      {
      trails.map((trail) => {
        let icon;
        switch (trail.difficulty) {
          case ('easy'):
            icon = require('../../assets/icons/triangle.jpg');
            break;
          case ('medium'):
            icon = require('../../assets/icons/rectangle.jpg');
            break;
          case ('hard'):
            icon = require('../../assets/icons/circle.jpg');
            break;
          default:
            break;
        }
        return (
          <TouchableOpacity
            key={trail.name}
            activeOpacity={0.5}
            onPress={() => handleSelectedTrail(trail.id)}
          >
            <ListItem
              avatar={icon}
              title={trail.name}
              subtitle={`${trail.length} miles`}
            />
          </TouchableOpacity>
        );
      })
      }
    </List>
  </ScrollView>
);

const styles = StyleSheet.create({
  list: {
    marginTop: 0,
    borderTopWidth: 0,
    borderBottomWidth: 0,
  },
});


TrailList.propTypes = {
  trails: PropTypes.arrayOf(PropTypes.object),
  handleSelectedTrail: PropTypes.func,
};

export default TrailList;
