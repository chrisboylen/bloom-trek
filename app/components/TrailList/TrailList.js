import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import { List, ListItem } from 'react-native-elements';
import PropTypes from 'prop-types';

const TrailList = ({ trails, handleSelectedTrail }) => (
  <ScrollView>
    <List containerStyle={styles.list} >
      {
      trails.map((trail) => {
        let icon;
        switch (trail.difficulty) {
          case ('easy'):
            icon = require('../../../assets/icons/triangle.jpg');
            break;
          case ('medium'):
            icon = require('../../../assets/icons/rectangle.jpg');
            break;
          case ('hard'):
            icon = require('../../../assets/icons/circle.jpg');
            break;
          default:
            break;
        }
        return (
          <ListItem
            avatar={icon}
            key={trail.name}
            title={trail.name}
            subtitle={`${trail.length} miles`}
            onPress={() => handleSelectedTrail(trail.id)}
          />
        );
      })
      }
    </List>
  </ScrollView>
);

const styles = StyleSheet.create({
  list: {
    marginTop: 0,
    borderTopWidth: 0,
    borderBottomWidth: 0,
  },
});


TrailList.propTypes = {
  trails: PropTypes.arrayOf(PropTypes.object),
  handleSelectedTrail: PropTypes.func,
};

export default TrailList;
