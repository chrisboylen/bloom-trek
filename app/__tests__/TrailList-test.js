import React from 'react';
import { shallow } from 'enzyme';
import TrailList from '../components/TrailList';

describe('TrailList', () => {
  let wrapper;
  let mockTrails;
  let mockHandleSelectedTrail;

  beforeEach(() => {
    mockTrails = [{
      id: 1,
      name: 'Castlewood Canyon - Rimrock to Creek Bottom Loop',
      uid: 7006769,
      difficulty: 'medium',
      length: 4.6,
      status: 'All Clear',
      trail_img_url: 'https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg',
      trail_url: 'https://www.hikingproject.com/trail/7006769/castlewood-canyon-rimrock-to-creek-bottom-loop',
      summary: 'A fun, rolling trail with a sense of seclusion, great geology, and historical sites.',
      park_id: 1,
      coords: {
        latitude: 39.3598,
        longitude: -104.7682,
      },
    }];
    mockHandleSelectedTrail = jest.fn();
    wrapper = shallow(
      <TrailList
        trails={mockTrails}
        handleSelectedTrail={mockHandleSelectedTrail}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should should invoke handleSelectedTrail onPress', () => {
    wrapper.find('TouchableOpacity').props().onPress();

    expect(mockHandleSelectedTrail).toHaveBeenCalled();
  });
});
