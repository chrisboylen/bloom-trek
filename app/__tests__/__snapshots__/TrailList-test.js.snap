// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrailList should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrailList
    handleSelectedTrail={[MockFunction]}
    trails={
      Array [
        Object {
          "coords": Object {
            "latitude": 39.3598,
            "longitude": -104.7682,
          },
          "difficulty": "medium",
          "id": 1,
          "length": 4.6,
          "name": "Castlewood Canyon - Rimrock to Creek Bottom Loop",
          "park_id": 1,
          "status": "All Clear",
          "summary": "A fun, rolling trail with a sense of seclusion, great geology, and historical sites.",
          "trail_img_url": "https://cdn-files.apstatic.com/hike/7006325_medium_1438452681.jpg",
          "trail_url": "https://www.hikingproject.com/trail/7006769/castlewood-canyon-rimrock-to-creek-bottom-loop",
          "uid": 7006769,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <List
        containerStyle={
          Object {
            "borderBottomWidth": 0,
            "borderTopWidth": 0,
            "marginTop": 0,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={[Function]}
        >
          <ListItem
            avatar={1}
            chevronColor="#bdc6cf"
            disabled={false}
            hideChevron={false}
            leftIconUnderlayColor="white"
            rightIcon={
              Object {
                "name": "chevron-right",
              }
            }
            rightTitleNumberOfLines={1}
            roundAvatar={false}
            subtitle="4.6 miles"
            subtitleNumberOfLines={1}
            switchButton={false}
            textInputEditable={true}
            title="Castlewood Canyon - Rimrock to Creek Bottom Loop"
            titleNumberOfLines={1}
            underlayColor="white"
          />
        </TouchableOpacity>
      </List>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TouchableOpacity
            activeOpacity={0.5}
            onPress={[Function]}
          >
            <ListItem
              avatar={1}
              chevronColor="#bdc6cf"
              disabled={false}
              hideChevron={false}
              leftIconUnderlayColor="white"
              rightIcon={
                Object {
                  "name": "chevron-right",
                }
              }
              rightTitleNumberOfLines={1}
              roundAvatar={false}
              subtitle="4.6 miles"
              subtitleNumberOfLines={1}
              switchButton={false}
              textInputEditable={true}
              title="Castlewood Canyon - Rimrock to Creek Bottom Loop"
              titleNumberOfLines={1}
              underlayColor="white"
            />
          </TouchableOpacity>,
        ],
        "containerStyle": Object {
          "borderBottomWidth": 0,
          "borderTopWidth": 0,
          "marginTop": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Castlewood Canyon - Rimrock to Creek Bottom Loop",
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.5,
            "children": <ListItem
              avatar={1}
              chevronColor="#bdc6cf"
              disabled={false}
              hideChevron={false}
              leftIconUnderlayColor="white"
              rightIcon={
                Object {
                  "name": "chevron-right",
                }
              }
              rightTitleNumberOfLines={1}
              roundAvatar={false}
              subtitle="4.6 miles"
              subtitleNumberOfLines={1}
              switchButton={false}
              textInputEditable={true}
              title="Castlewood Canyon - Rimrock to Creek Bottom Loop"
              titleNumberOfLines={1}
              underlayColor="white"
            />,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "avatar": 1,
              "chevronColor": "#bdc6cf",
              "disabled": false,
              "hideChevron": false,
              "leftIconUnderlayColor": "white",
              "rightIcon": Object {
                "name": "chevron-right",
              },
              "rightTitleNumberOfLines": 1,
              "roundAvatar": false,
              "subtitle": "4.6 miles",
              "subtitleNumberOfLines": 1,
              "switchButton": false,
              "textInputEditable": true,
              "title": "Castlewood Canyon - Rimrock to Creek Bottom Loop",
              "titleNumberOfLines": 1,
              "underlayColor": "white",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <List
          containerStyle={
            Object {
              "borderBottomWidth": 0,
              "borderTopWidth": 0,
              "marginTop": 0,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.5}
            onPress={[Function]}
          >
            <ListItem
              avatar={1}
              chevronColor="#bdc6cf"
              disabled={false}
              hideChevron={false}
              leftIconUnderlayColor="white"
              rightIcon={
                Object {
                  "name": "chevron-right",
                }
              }
              rightTitleNumberOfLines={1}
              roundAvatar={false}
              subtitle="4.6 miles"
              subtitleNumberOfLines={1}
              switchButton={false}
              textInputEditable={true}
              title="Castlewood Canyon - Rimrock to Creek Bottom Loop"
              titleNumberOfLines={1}
              underlayColor="white"
            />
          </TouchableOpacity>
        </List>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TouchableOpacity
              activeOpacity={0.5}
              onPress={[Function]}
            >
              <ListItem
                avatar={1}
                chevronColor="#bdc6cf"
                disabled={false}
                hideChevron={false}
                leftIconUnderlayColor="white"
                rightIcon={
                  Object {
                    "name": "chevron-right",
                  }
                }
                rightTitleNumberOfLines={1}
                roundAvatar={false}
                subtitle="4.6 miles"
                subtitleNumberOfLines={1}
                switchButton={false}
                textInputEditable={true}
                title="Castlewood Canyon - Rimrock to Creek Bottom Loop"
                titleNumberOfLines={1}
                underlayColor="white"
              />
            </TouchableOpacity>,
          ],
          "containerStyle": Object {
            "borderBottomWidth": 0,
            "borderTopWidth": 0,
            "marginTop": 0,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Castlewood Canyon - Rimrock to Creek Bottom Loop",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.5,
              "children": <ListItem
                avatar={1}
                chevronColor="#bdc6cf"
                disabled={false}
                hideChevron={false}
                leftIconUnderlayColor="white"
                rightIcon={
                  Object {
                    "name": "chevron-right",
                  }
                }
                rightTitleNumberOfLines={1}
                roundAvatar={false}
                subtitle="4.6 miles"
                subtitleNumberOfLines={1}
                switchButton={false}
                textInputEditable={true}
                title="Castlewood Canyon - Rimrock to Creek Bottom Loop"
                titleNumberOfLines={1}
                underlayColor="white"
              />,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "avatar": 1,
                "chevronColor": "#bdc6cf",
                "disabled": false,
                "hideChevron": false,
                "leftIconUnderlayColor": "white",
                "rightIcon": Object {
                  "name": "chevron-right",
                },
                "rightTitleNumberOfLines": 1,
                "roundAvatar": false,
                "subtitle": "4.6 miles",
                "subtitleNumberOfLines": 1,
                "switchButton": false,
                "textInputEditable": true,
                "title": "Castlewood Canyon - Rimrock to Creek Bottom Loop",
                "titleNumberOfLines": 1,
                "underlayColor": "white",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
